sip/lib/sip_endpoint.ex

```diff
@@ -0,0 +1,731 @@
+          OutgoingCall.cancel(state.call_id)
+          :ending_call
+
+        :in_call ->
+          Logger.info("SIP Endpoint: Ending call")
+          OutgoingCall.bye(state.call_id)
+          :ending_call
+
+        other_state ->
+          Logger.warning(
+            "SIP Endpoint: No ongoing call or call attempt to end, or endpoint is already terminating"
+          )
+
+          other_state
+      end
+
+    {[], %{state | phone_number: nil, endpoint_state: new_endpoint_state}}
+  end
+
+  @impl true
+  def handle_parent_notification({:new_tracks, tracks}, ctx, state) do
+    {:endpoint, endpoint_id} = ctx.name
+
+    state =
+      tracks
+      |> Enum.filter(fn track -> track.type == :audio end)
+      |> Enum.reduce(state, fn track, state ->
+        case Engine.subscribe(state.rtc_engine, endpoint_id, track.id) do
+          :ok ->
+            put_in(state, [:incoming_tracks, track.id], track)
+
+          {:error, :invalid_track_id} ->
+            Logger.info("""
+            Couldn't subscribe to the track: #{inspect(track.id)}. No such track.
+            It had to be removed just after publishing it. Ignoring.
+            """)
+
+            state
+
+          {:error, reason} ->
+            raise "Couldn't subscribe to the track: #{inspect(track.id)}. Reason: #{inspect(reason)}"
+        end
+      end)
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_parent_notification({topic, _data}, _ctx, state)
+      when topic in [:remove_tracks, :ready, :endpoint_added, :endpoint_removed] do
+    {[], state}
+  end
+
+  @impl true
+  def handle_parent_notification(%TrackNotification{}, _ctx, state) do
+    {[], state}
+  end
+
+  @impl true
+  def handle_parent_notification(msg, _ctx, state) do
+    Logger.warning("SIP Endpoint: Unexpected message from parent: #{inspect(msg)}. Ignoring.")
+    {[], state}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:new_rtp_stream, ssrc, fmt, _extensions} = msg,
+        :rtp,
+        _ctx,
+        state
+      )
+      when is_nil(state.first_ssrc) or ssrc == state.first_ssrc do
+    Logger.debug("SIP Endpoint: New RTP stream connected: #{inspect(msg)}")
+
+    state = %{state | first_ssrc: ssrc}
+
+    if fmt != state.payload_type do
+      raise """
+      Payload type mismatch between RTP mapping and received stream
+      (expected #{inspect(state.payload_type)}, got #{inspect(fmt)})
+      """
+    end
+
+    {[
+       notify_child: {:audio_mixer, {:start_mixing, @audio_mixer_delay}},
+       notify_parent:
+         {:track_ready, state.outgoing_track.id, :high, state.outgoing_track.encoding},
+       notify_parent: {:forward_to_parent, :received_rtp_stream}
+     ], state}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:new_rtp_stream, ssrc, fmt, _extensions},
+        :rtp,
+        _ctx,
+        state
+      ) do
+    Logger.warning("Received another RTP stream with ssrc: #{ssrc}")
```

Comment:
I think we could decrease the level of this log to info as, after reading this line, I don't know about what this warns me.